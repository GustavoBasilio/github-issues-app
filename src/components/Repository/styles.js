import styled from 'styled-components';
import { white, grayDarker, grayDark } from '../../constants/colors';

export const Container = styled.View`
    background-color: ${white}
    border-radius: 6px;
    display: flex;
    flex-direction: row;
    padding: 20px 20px;
    margin-bottom: 10px;
`;

export const Avatar = styled.Image`
    width: 35px;
    height: 35px;
    justify-content: center;
    padding:  0 10px;
`;

export const Info = styled.View`
    display: flex;
    flex-grow: 1;
    flex-direction: column;
    padding-left: 20px;
`;

export const Name = styled.Text`
    color: ${grayDarker};
    font-size: 18px;
`;

export const Owner = styled.Text`
    color: ${grayDark};
    font-size: 14px;
`;


export const Submit = styled.View`
    justify-content: center;
`;

