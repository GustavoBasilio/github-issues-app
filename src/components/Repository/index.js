import React from 'react';
import Icon from 'react-native-vector-icons/MaterialIcons';

import { grayDark } from '../../constants/colors';
import { Container, Avatar, Info, Name, Owner, Submit} from './styles';

const Repository = ({data}) => {
    return (
        <Container>
            <Avatar source={{uri: data.avatar}} />
            <Info>
                <Name>{data.name}</Name>
                <Owner>{data.owner}</Owner>
            </Info>
            <Submit>
                <Icon name="keyboard-arrow-right" size={30} color={grayDark} />
            </Submit>
        </Container>
    )
};

export default Repository;
