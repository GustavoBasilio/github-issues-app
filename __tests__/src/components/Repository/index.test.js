/**
 * @format
 */

import 'react-native';
import React from 'react';
import Repository from '../../../../src/components/Repository/index';

import 'jest-styled-components'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const comp = renderer.create(<Repository data={{
    "id": "",
    "name": "",
    "owner": "",
    "avatar": ""
}
} />).toJSON();
  expect(comp).toMatchSnapshot();
});
