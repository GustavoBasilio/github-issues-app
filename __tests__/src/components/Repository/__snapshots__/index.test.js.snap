// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 6,
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 10,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Array [
        Object {
          "height": 35,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 0,
          "width": 35,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": 1,
          "paddingLeft": 20,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#373737",
            "fontSize": 18,
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#E5E5E5",
            "fontSize": 14,
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#E5E5E5",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      îŒ•
    </Text>
  </View>
</View>
`;
